types/qaengine/commonqa/commonqa.go

```diff
@@ -76,3 +79,45 @@ func IngressHost(defaulthost string) string {
 func MinimumReplicaCount(defaultminreplicas string) string {
 	return qaengine.FetchStringAnswer(common.ConfigMinReplicasKey, "Provide the minimum number of replicas each service should have", []string{"If the value is 0 pods won't be started by default"}, defaultminreplicas)
 }
+
+// GetPortsForService returns ports used by a service
+func GetPortsForService(detectedPorts []int32) []int32 {
+	var selectedPortsStr, enteredPortsStr, detectedPortsStr []string
+	var enteredPorts string
+	var exposePorts []int32
+	if len(detectedPorts) != 0 {
+		for _, detectedPort := range detectedPorts {
+			detectedPortsStr = append(detectedPortsStr, strconv.Itoa(int(detectedPort)))
+		}
+		allDetectedPortsStr := append(detectedPortsStr, qatypes.OtherAnswer)
+		selectedPortsStr = qaengine.FetchMultiSelectAnswer(common.ConfigServicesKey+common.Delim+common.ConfigPortsForServiceKeySegment, "Select ports to be exposed :", []string{"Select Other if you want to add more ports"}, detectedPortsStr, allDetectedPortsStr)
+	}
+	if len(selectedPortsStr) == 0 || common.IsStringPresent(selectedPortsStr, qatypes.OtherAnswer) {
+		enteredPorts = qaengine.FetchMultilineAnswer(common.ConfigServicesKey+common.Delim+common.ConfigAdditionalPortsForServiceKeySegment, "Enter the ports to be exposed", []string{"Enter each port in a newline"}, "")
+	}
+
+	enteredPortsStr = strings.Split(enteredPorts, "\n")
+	for _, portStr := range selectedPortsStr {
+		if portStr != qatypes.OtherAnswer {
```

Comment:
if portStr -= qatypes.OtherAnswer {
continue
}

will help reduce the indentations

Reply:
Yes sure, thanks.